src/emm/clients/git_proxy.py

```diff
@@ -308,10 +313,79 @@ def push_branch_to_remote(self, directory: Optional[Path] = None) -> str:
                 f"in directory '{self._determine_directory(directory)}'"
             )
 
-        args = ["push", "-u", "origin", "HEAD"]
+        args = ["push", "-u", remote, "HEAD"]
         with local.cwd(self._determine_directory(directory)):
             return self.git[args]().strip()
 
+    def current_remote(self, directory: Optional[Path] = None) -> List[str]:
+        """
+        Get the current remote and associate remote repo url.
+
+        :param directory: Directory to execute command at.
+        :return: List of remote names and remote urls.
+        """
+        args = ["config", "--get-regexp", ".url"]
```

Comment:
This seems like a convoluted way to get a remote url.

Why not something like: `git remote get-url origin`?